packages/headless/src/test/mock-result.ts

```diff
@@ -1,7 +1,14 @@
 import {Result} from '../api/search/search/result';
 import {buildMockRaw} from './mock-raw';
 
-export function buildMockResult(config: Partial<Result> = {}): Result {
+interface ResultWithFolding extends Result {
+  parentResult?: ResultWithFolding;
+  childResults?: ResultWithFolding[];
+}
```

Comment:
Let's create a dedicated file for the foldedResult mock. We should also reuse the interface from the source code rather than redeclare one here,